spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java

```diff
@@ -0,0 +1,175 @@
+          16.0, 20.0, 25.0, 30.0, 40.0, 50.0, 65.0, 80.0, 100.0, 130.0, 160.0, 200.0, 250.0,
+          300.0, 400.0, 500.0, 650.0, 800.0, 1000.0, 2000.0, 5000.0, 10000.0, 20000.0, 50000.0,
+          100000.0)));
+  static final View GFE_LATENCY_VIEW = View
+      .create(Name.create("cloud.google.com/java/spanner/gfe_latency"),
+          "Latency between Google's network receives an RPC and reads back the first byte of the"
+              + " response",
+          SPANNER_GFE_LATENCY,
+          AGGREGATION_WITH_MILLIS_HISTOGRAM,
+          Collections.singletonList(key));
+
+  static ViewManager manager = Stats.getViewManager();
+
+  private static final Tagger tagger = Tags.getTagger();
+  private static final StatsRecorder STATS_RECORDER = Stats.getStatsRecorder();
+
+  static void captureGfeMetric(DatabaseClient dbClient) {
+    // Register GFE view.
+    manager.registerView(GFE_LATENCY_VIEW);
+
+    // Enable OpenCensus exporters to export metrics to Stackdriver Monitoring.
+    // Exporters use Application Default Credentials to authenticate.
+    // See https://developers.google.com/identity/protocols/application-default-credentials
+    // for more details.
+    try {
+      StackdriverStatsExporter.createAndRegister();
+    } catch (IOException | IllegalStateException e) {
+      System.out.println("Error during StackdriverStatsExporter");
+    }
+
+    try (ResultSet resultSet =
+        dbClient
+            .singleUse() // Execute a single read or query against Cloud Spanner.
+            .executeQuery(Statement.of("SELECT SingerId, AlbumId, AlbumTitle FROM Albums"))) {
+      while (resultSet.next()) {
+        System.out.printf(
+            "%d %d %s", resultSet.getLong(0), resultSet.getLong(1), resultSet.getString(2));
+      }
+    }
+  }
+
+  private static final HeaderClientInterceptor interceptor = new HeaderClientInterceptor();
+  private static final Metadata.Key<String> SERVER_TIMING_HEADER_KEY =
+      Metadata.Key.of("server-timing", Metadata.ASCII_STRING_MARSHALLER);
+  private static final Pattern SERVER_TIMING_HEADER_PATTERN = Pattern.compile(".*dur=(?<dur>\\d+)");
+
+  // ClientInterceptor to capture GFE header latency.
+  private static class HeaderClientInterceptor implements ClientInterceptor {
+
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method,
+        CallOptions callOptions, Channel next) {
+      return new SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+
+        @Override
+        public void start(Listener<RespT> responseListener, Metadata headers) {
+          super.start(new SimpleForwardingClientCallListener<RespT>(responseListener) {
+            @Override
+            public void onHeaders(Metadata metadata) {
+              processHeader(metadata, method.getFullMethodName());
+              super.onHeaders(metadata);
+            }
+          }, headers);
+        }
+      };
+    }
```

Comment:
OPTIONAL - consider explaining this or provide a link to the docs.  If it's not already on the page you display this on..